modules/product/src/main/java/com/opengamma/strata/product/credit/CdsIndex.java

```diff
@@ -0,0 +1,1220 @@
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.BusinessDayAdjustment;
+import com.opengamma.strata.basics.date.BusinessDayConventions;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.date.DayCounts;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.basics.date.HolidayCalendarId;
+import com.opengamma.strata.basics.schedule.Frequency;
+import com.opengamma.strata.basics.schedule.PeriodicSchedule;
+import com.opengamma.strata.basics.schedule.RollConventions;
+import com.opengamma.strata.basics.schedule.Schedule;
+import com.opengamma.strata.basics.schedule.SchedulePeriod;
+import com.opengamma.strata.basics.schedule.StubConvention;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.common.BuySell;
+
+/**
+ * A CDS (portfolio) index product. 
+ * <p>
+ * A CDS index is a portofolio of single name credit default swaps. 
+ * The contract periodically pays fixed coupons to the index buyer until the expiry, 
+ * and in return, the index buyer receives the bond of a defaulted constituent legal entity for par.  
+ */
+@BeanDefinition
+public final class CdsIndex
+    implements Resolvable<ResolvedCdsIndex>, ImmutableBean, Serializable {
+
+  /**
+   * Whether the CDS index is buy or sell.
+   * <p>
+   * A value of 'Buy' implies buying credit risk, where the fixed coupon is received
+   * and the protection is paid  in the event of default.
+   * A value of 'Sell' implies selling credit risk, where the fixed coupon is paid
+   * and the protection is received in the event of default. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final BuySell buySell;
+  /**
+   * The CDS index identifier.
+   * <p>
+   * This identifier is used for referring this CDS index product.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId cdsIndexId;
+  /**
+   * The legal entity identifiers.
+   * <p>
+   * This identifiers are used for the reference legal entities of the CDS index.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableList<StandardId> legalEntityIds;
+  /**
+   * The currency of the CDS index.
+   * <p>
+   * The amounts of the notional are expressed in terms of this currency.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final Currency currency;
+  /**
+   * The notional amount, must be non-negative.
+   * <p>
+   * The fixed notional amount applicable during the lifetime of the CDS.
+   * The currency of the notional is specified by {@code currency}.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegativeOrZero")
+  private final double notional;
+  /**
+   * The accrual schedule.
+   * <p>
+   * This is used to define the accrual periods.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final PeriodicSchedule accrualSchedule;
```

Comment:
Perhaps this should be called `paymentSchedule` (with a similar change on `Cds`)? (swaps have separate accrual and payment schedules, but CDS does not.)